chore: Document how to use form controls in a Field (Set) (#1548)
Amsterdam/design-system: storybook/src/components/Radio/Radio.docs.mdx

<Controls />

## Examples

### In a Field Set

Use a Field Set to group several Radios with a legend, description and / or an error message.

> When grouping Radios, add `role="radiogroup"` to the Field Set to have it explicitly announced as a radio group (the default role is just ‘group’).

Reviewer: No need to repeat the grouping once more.

```suggestion
Add `role="radiogroup"` to the Field Set to have it explicitly announced as a radio group (the default role is just ‘group’).
```

The ‘radio group’ role also allows using `aria-required` on Field Set; that isn’t allowed for the ‘group’ role.
Always also set `aria-required` on the individual Radios though, to make sure it’s read by screen readers.

> If you don’t need the description, just take out the description Paragraph and the `aria-describedby` from the Field Set.

Reviewer: ```suggestion
If you don’t need the description, remove its Paragraph and the `aria-describedby` from the Field Set.
```

<Controls />

## Examples

### In a Field Set

Usually, you should place Radio’s inside a Field Set component.

> Here’s an example of a Field Set component that includes a legend, a description, and a group of Radio’s.

Reviewer: This is more to the point – is it strong enough to combine these two lines?

> Use a Field Set to group several Radios with a label, description, and error message.

When grouping radio inputs, use `role="radiogroup"` on Field Set to have this grouping explicitly announced as a radio group (the default role is group).
Using `role="radiogroup"` also allows you to use `aria-required` on Field Set, which isn’t allowed for role group.
Always also set `aria-required` on the individual radio buttons though, to make sure it’s read by screen readers.

If you don’t need the description, just take out the description Paragraph and the `aria-describedby` from the Field Set.

> For more information, check [the Field Set component documentation](/docs/components-forms-field-set--docs).

Reviewer: ```suggestion
Check [Field Set](/docs/components-forms-field-set--docs) for more information on configuring it.
```

<Canvas of={RadioStories.InAFieldSet} />

### In an invalid Field Set

> This is an example of an invalid Field Set with a Label, a description, an Error Message and a group of Radio’s.

Reviewer: ```suggestion
If the Radio can become invalid, add an Error Message component and its `id` to the `aria-describedby` prop of the Field Set.
```
